### PR TITLE
[Android] Implement GitHub Action to test Pull Requests

### PR DESCRIPTION
**Pull request by IanDBird**
This PR adds a new GitHub Action that should detect changes to any of the Android based sample sites, and try to build them. Hopefully ensuring that we don't land any changes that break the build.

### CODE REVIEW

File: .github/workflows/build-android.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Test Android Pull Requests
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'android/**'
```
Comment (mrmans0n) [inline .github/workflows/build-android.yml:7]: You should probably also run this on `push` on main, with same path filter in it.

Comment (IanDBird) [reply .github/workflows/build-android.yml:7]: Done, thanks :)

File: .github/workflows/build-android.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: Test Android Pull Requests
+
+on:
+  pull_request:
+    branches: [ main ]
```
Comment (koalaparadise) [inline .github/workflows/build-android.yml:5]: Remove the `branches` block here.

File: .github/workflows/build-android.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Grant execute permission for gradlew (TV Custom)
+        working-directory: android/tv-custom
+        run: chmod +x gradlew
+
+      - name: Grant execute permission for gradlew (TV Device Auth Grant)
+        working-directory: android/tv-device-auth-grant
+        run: chmod +x gradlew
+
+      - name: Build with Gradle (Mobile)
```
Comment (koalaparadise) [inline .github/workflows/build-android.yml:45]: Use the `setup-gradle` action. It adds some nice caching.

https://github.com/gradle/actions?tab=readme-ov-file#the-setup-gradle-action

File: .github/workflows/build-android.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew (Mobile)
```
Comment (koalaparadise) [inline .github/workflows/build-android.yml:33]: FYI, you can fix the `chmod` issue locally and upload it in the PR. File permissions are stored in git

File: .github/workflows/build-android.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
```
Comment (koalaparadise) [inline .github/workflows/build-android.yml:31]: Remove this once you use `setup-gradle` below

**Review by koalaparadise — APPROVED** [review]

**Pull request merged**